tests/integrationtest/lib.go

```diff
@@ -952,6 +955,88 @@ func DelSVC(t *testing.T, ns string, Name string) {
 	}
 }
 
+// CreateEPorEPS creates endpoint or endpoint slices based on env
+func CreateEPorEPS(t *testing.T, ns string, Name string, multiPort bool, multiAddress bool, addressPrefix string, multiProtocol ...corev1.Protocol) {
+	if !lib.AKOControlConfig().GetEndpointSlicesEnabled() {
+		CreateEP(t, ns, Name, multiPort, multiAddress, addressPrefix, multiProtocol...)
+		return
+	}
+	addressType := discovery.AddressTypeIPv4
+
+	if addressPrefix == "" {
+		addressPrefix = "1.1.1"
+	}
+	if strings.Contains(addressPrefix, "::") {
+		addressType = discovery.AddressTypeIPv6
+	}
+	//var endpoints discovery.EndpointSlice
+	numPorts, numAddresses := 1, 1
+	if multiPort {
+		numPorts = 3
+	}
+	if len(multiProtocol) != 0 {
+		numPorts = len(multiProtocol)
+	}
+	if multiAddress {
+		numAddresses = 3
+	}
+	svcName := Name
+
+	// // create separate endpoint slice if multiProtocol
+
+	// // mPort := 8080 + i
+	startIndex := 0
+	for i := 0; i < numPorts; i++ {
+		var endpoints []discovery.Endpoint
+		for j := 0; j < numAddresses; j++ {
+			if strings.Contains(addressPrefix, "::") {
+				endpoints = append(endpoints, discovery.Endpoint{
+					Addresses: []string{fmt.Sprintf("%s%d", addressPrefix, j+startIndex+1)},
+				})
+			} else {
+				endpoints = append(endpoints, discovery.Endpoint{Addresses: []string{fmt.Sprintf("%s.%d", addressPrefix, j+startIndex+1)}})
+			}
+			startIndex++
+		}
+		numAddresses--
+		protocol := corev1.ProtocolTCP
+		if len(multiProtocol) != 0 {
+			protocol = multiProtocol[i]
+		}
+		mPort := int32(8080 + i)
+		portName := fmt.Sprintf("foo%d", i)
+		ports := []discovery.EndpointPort{{
+			Protocol: &protocol,
+			Port:     &mPort,
+			Name:     &portName,
+		}}
+		tempName := Name + "-" + randStringBytesRmndr(5)
+		epExample := &discovery.EndpointSlice{
+			AddressType: addressType,
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+				Name:      tempName,
+				Labels:    map[string]string{discovery.LabelServiceName: svcName},
+			},
+			Endpoints: endpoints,
+			Ports:     ports,
+		}
+		_, err := KubeClient.DiscoveryV1().EndpointSlices(ns).Create(context.TODO(), epExample, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("error in creating Endpoint: %v", err)
+		}
+
+	}
+	time.Sleep(2 * time.Second)
```

Comment:
Explicit sleep is require over here?

Reply:
Yes, this was also added in CreateEP at L1119 previously